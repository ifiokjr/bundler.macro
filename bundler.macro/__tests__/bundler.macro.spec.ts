import plugin from 'babel-plugin-macros';
import pluginTester from 'babel-plugin-tester';
import path from 'path';

const fixtures = (...paths: string[]) => path.join(__dirname, '..', '__fixtures__', ...paths);

pluginTester({
  plugin,
  pluginName: 'bundler.macro',
  title: 'transpileFile',
  snapshot: true,
  babelOptions: {
    filename: fixtures('file.js'),
  },

  tests: {
    'run correctly': {
      code: `import { transpileFile } from './macro.js';
      const output = transpileFile('./transpile.fixture.js');`,
      snapshot: true,
    },
  },
});

pluginTester({
  plugin,
  pluginName: 'bundler.macro',
  title: 'rollupBundle',
  snapshot: true,
  babelOptions: {
    filename: fixtures('file.ts'),
  },

  tests: {
    'run correctly': {
      code: `import { rollupBundle } from './macro.js';
      const output = rollupBundle('./bundle.fixture.js');`,
      snapshot: true,
    },
    'bundles packages with node included': {
      code: `import { rollupBundle } from './macro.js';
      const output = rollupBundle('./react-dom.fixture.js');`,
      snapshot: true,
    },
  },
});

pluginTester({
  plugin,
  pluginName: 'bundler.macro',
  title: 'esbuildBundle',
  snapshot: true,
  babelOptions: {
    filename: fixtures('file.ts'),
  },

  tests: {
    'run correctly': {
      code: `import { esbuildBundle } from './macro.js';
      const output = esbuildBundle('./bundle.fixture.js');`,
      snapshot: true,
    },
    'bundles packages with node included': {
      code: `import { esbuildBundle } from './macro.js';
      const output = esbuildBundle('./react-dom.fixture.js');`,
      snapshot: true,
    },
  },
});
