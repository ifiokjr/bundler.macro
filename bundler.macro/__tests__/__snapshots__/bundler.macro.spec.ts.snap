// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollupBundle run correctly: run correctly 1`] = `

import { rollupBundle } from './macro.js';
      const output = rollupBundle('./bundle.fixture.js');

      ↓ ↓ ↓ ↓ ↓ ↓

const output =
  "var $$_MACRO_BUNDLER_GLOBAL_$$ = (function (exports) {\\n  'use strict';\\n\\n  function fn() {\\n    return 'awesome';\\n  }\\n\\n  const a = /*#__PURE__*/fn();\\n\\n  exports.a = a;\\n  exports.default = fn;\\n\\n  Object.defineProperty(exports, '__esModule', { value: true });\\n\\n  return exports;\\n\\n}({}));\\n";


`;

exports[`transpileFile run correctly: run correctly 1`] = `

import { transpileFile } from './macro.js';
      const output = transpileFile('./transpile.fixture.js');

      ↓ ↓ ↓ ↓ ↓ ↓

const output = "function TranspiledJS() {\\n  return 'this should be transpiled';\\n}";


`;
